src/main/java/org/jenkinsci/plugins/vsphere/builders/Deploy.java

```diff
@@ -318,7 +318,7 @@ private void add(String key, String val) {
         }
 
         @Override
-        public void buildEnvVars(AbstractBuild<?,?> build, EnvVars env) {
+        public void buildEnvironment(@Nonnull Run<?, ?> run, @Nonnull EnvVars env) {
```

Comment:
This set of changes worries me.
There's a lot of tech-debt buried in these classes and, while it would be nice to resolve all of that, the testing effort to ensure that it still all works is non-trivial.
Best do that in a separate PR.

Reply:
Again targeting newer Jenkins version this is the correct, the base class converts older AbstractBuild to Run for you behind the scene.

Reply:
OK.  There's a fair amount of code duplication tech-debt in this area and it'd be nice to get that cleaned up ... but lets do that in a separate PR after bumping the Java version and minimum Jenkins version.

Reply:
Reverted